core/sql/executor/ExHdfsScan.cpp

```diff
@@ -1700,6 +1714,138 @@ char * ExHdfsScanTcb::extractAndTransformAsciiSourceToSqlRow(int &err,
   return NULL;
 }
 
+void ExHdfsScanTcb::computeRangesAtRuntime()
```

Comment:
Yes, but doing so would be quite complex, since we would have to send the data to the EPSs. Your proposal is somewhat related to Dave's proposal to generate the temp table name at runtime - this would also require sending information to the ESPs. So, maybe we can consider both of those together. Because of the security holes with Hive tables, however, I'm wondering whether we should try volatile tables first before making the Hive table solution better.